compiler/rustc_borrowck/src/diagnostics/bound_region_errors.rs

```diff
@@ -172,20 +179,16 @@ pub(crate) trait TypeOpInfo<'tcx> {
             ty::Placeholder { universe: adjusted_universe.into(), bound: placeholder.bound },
         );
 
-        let error_region = if let RegionElement::PlaceholderRegion(error_placeholder) =
-            error_element
-        {
-            let adjusted_universe =
-                error_placeholder.universe.as_u32().checked_sub(base_universe.as_u32());
```

Comment:
I think the "checked sub" part is what really gets me here. Why are universes being adjusted? What's a base universe? And why can the subtraction sometimes overflow?!?!